frontend/packages/ui/src/ProgressBar.tsx

```diff
@@ -0,0 +1,68 @@
+import React from 'react'
+import { useState, useEffect } from 'react'
+import { BarChart, Bar, XAxis, YAxis, Tooltip, ResponsiveContainer } from 'recharts'
+import { Colors } from './styles/colors'
+
+interface Props {
+  data: { name: string; size: number; color: string }[]
+  onClick: () => void
+}
+
+const HEIGHT = 20
+
+export const ProgressBar = (props: Props) => {
+
+  const [barData, setBarData] = useState(null)
+  const [barColors, setBarColors] = useState(null)
+
+  const prepareBarChartValues = () => {
```

Comment:
We don't need to use useState we can just manipulate data directly:
`const barData = props.data.map(....)`
 and also if props.data changes outside of the component this will not trigger again, because useEffect has no dependancies.

Reply:
I added dependencies. But I was not able to do this without useState.
If it is necessary to avoid useState I will need further instructions